contracts/gateway/AxelarAmplifierGateway.sol

```diff
@@ -63,93 +78,117 @@ contract AxelarAmplifierGateway is IAxelarAmplifierGateway {
         string calldata sourceAddress,
         bytes32 payloadHash
     ) external override returns (bool valid) {
-        bytes32 key = _getIsContractCallApprovedKey(commandId, sourceChain, sourceAddress, msg.sender, payloadHash);
-        valid = _axelarAmplifierGatewayStorage().approvals[key];
-
-        if (valid) {
-            delete _axelarAmplifierGatewayStorage().approvals[key];
+        valid = _validateContractCall(commandId, sourceChain, sourceAddress, payloadHash);
```

Comment:
This used to revert, are we sure we want to change the pattern here (`AxelarExecutable` will have to change to account for this as well)

Reply:
we don't revert in the existing gateway
https://github.com/axelarnetwork/axelar-cgp-solidity/blob/main/contracts/AxelarGateway.sol#L239

Reply:
Oh I must have had it wrong in my head, this seems good then.